sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala

```diff
@@ -61,8 +61,8 @@ class SparkOptimizer(
       new RowLevelOperationRuntimeGroupFiltering(OptimizeSubqueries)),
     Batch("InjectRuntimeFilter", FixedPoint(1),
       InjectRuntimeFilter),
-    Batch("MergeScalarSubqueries", Once,
-      MergeScalarSubqueries,
+    Batch("MergeSubplans", Once,
+      MergeSubplans,
```

Comment:
While looking at the code, I become curious. Can we have this improvement as a new rule, @peter-toth ?